src/openzaak/components/autorisaties/api/viewsets.py

```diff
@@ -104,10 +112,36 @@ class ApplicatieViewSet(
 
     queryset = (
         Applicatie.objects.exclude(
-            heeft_alle_autorisaties=False, autorisaties__isnull=True
+            Q(heeft_alle_autorisaties=False)
+            & ~Q(autorisaties__isnull=False)
+            & ~Q(catalogusautorisatie__isnull=False)
+        )
+        .exclude(
+            Q(heeft_alle_autorisaties=True)
+            & (Q(autorisaties__isnull=False) | Q(catalogusautorisatie__isnull=False))
+        )
+        .prefetch_related(
+            "autorisaties",
+            Prefetch(
+                "catalogusautorisatie_set",
+                queryset=CatalogusAutorisatie.objects.select_related("catalogus")
+                .order_by("component")
```

Comment:
Is this `order_by` needed?

Reply:
It's mostly for the tests if I remember correctly, I was running into issues with assertions because the ordering was unpredictable